src/core/server/http/http_service.ts

```diff
@@ -94,6 +99,23 @@ export class HttpService implements CoreService<HttpServiceSetup, HttpServiceSta
     };
   }
 
+  private async createServer(config: HttpConfig) {
+    const port = config.port;
+
+    if (!port || this.secondaryServers.has(port)) {
+      throw new Error(`port ${port} is already in use`);
```

Comment:
I'm not sure it is possible for `port` to not exist, so this check could probably be simplified to remove the negation check.

Reply:
Calling this from a javascript plugin could make `port` not exist here. While I enjoy typescript's type-safety functionality I still know how the sausage is being made and by default will try to protect non-ts consumers from shooting themselves in the foot.

Should I remove this then?

Reply:
No, good catch. I forgot that this could be called from JS land.